From b1a2e77fe171183b3fe82508136596bd7b66f339 Mon Sep 17 00:00:00 2001
From: Huang Chaofan <chaofan.huang@nxp.com>
Date: Thu, 13 Sep 2018 13:57:24 +0800
Subject: [PATCH 4655/5242] MLK-19568 VPU: Change the default frame depth to
 -1

commit  4bb0eafc0249c14afa154aecfe86a95eaa5b20d5 from
https://source.codeaurora.org/external/imx/linux-imx.git

Change the default frame depth to -1, to avoid that if input data is not
passed to driver by frame and it will block input data, and the frame depth
need to be set by user if user want it work

Signed-off-by: Huang Chaofan <chaofan.huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-decoder-b0/vpu_b0.c |    7 +++++--
 drivers/mxc/vpu-decoder-b0/vpu_b0.h |    1 +
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.c b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
index f5987c6..d59f468 100755
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.c
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
@@ -50,7 +50,7 @@
 #include "insert_startcode.h"
 
 unsigned int vpu_dbg_level_decoder = 1;
-static int vpu_frm_depth = 100;
+static int vpu_frm_depth = INVALID_FRAME_DEPTH;
 
 /* Generic End of content startcodes to differentiate from those naturally in the stream/file */
 #define EOS_GENERIC_HEVC 0x7c010000
@@ -1450,7 +1450,10 @@ static void v4l2_update_stream_addr(struct vpu_ctx *ctx, uint32_t uStrBufIdx)
 	uint32_t buffer_size;
 
 	down(&This->drv_q_lock);
-	while (!list_empty(&This->drv_q) && (ctx->frm_dec_delay < vpu_frm_depth)) {
+	while (!list_empty(&This->drv_q)) {
+		if (vpu_frm_depth != INVALID_FRAME_DEPTH) //frame depth need to be set by user and then the condition works
+			if (ctx->frm_dec_delay >= vpu_frm_depth)
+				break;
 		p_data_req = list_first_entry(&This->drv_q,
 				typeof(*p_data_req), list);
 
diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.h b/drivers/mxc/vpu-decoder-b0/vpu_b0.h
index 86f4bd4..9e35902 100755
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.h
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.h
@@ -66,6 +66,7 @@
 
 #define V4L2_MAX_CTRLS 12
 #define V4L2_PIX_FMT_NV12_10BIT    v4l2_fourcc('N', 'T', '1', '2') /*  Y/CbCr 4:2:0 for 10bit  */
+#define INVALID_FRAME_DEPTH -1
 
 struct vpu_v4l2_control {
 	uint32_t id;
-- 
1.7.9.5

